test/e2e/cluster_status_test.go

```diff
@@ -80,3 +85,53 @@ func TestBackupStatus(t *testing.T) {
 		t.Error(err)
 	}
 }
+
+// TestUpdateClusterAfterItFailed creates a failed cluster, then tries to update it,
+// and finally checks operator not panic.
+func TestUpdateClusterAfterItFailed(t *testing.T) {
+	f := framework.Global
+
+	cl := &spec.Cluster{
+		TypeMeta: unversioned.TypeMeta{
+			Kind:       strings.Title(spec.TPRKind),
+			APIVersion: spec.TPRGroup + "/" + spec.TPRVersion,
+		},
+		Metadata: v1.ObjectMeta{
+			GenerateName: "test-etcd-",
+		},
+		Spec: spec.ClusterSpec{
+			Size: 1,
+		},
+		Status: spec.ClusterStatus{
+			Phase: spec.ClusterPhaseFailed,
+		},
+	}
+	testEtcd, err := createCluster(t, f, cl)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		if err := deleteEtcdCluster(t, f, testEtcd); err != nil {
+			t.Fatal(err)
+		}
+	}()
+	testEtcd.Spec.Size = 3
+	if _, err := updateEtcdCluster(f, testEtcd); err != nil {
+		t.Fatal(err)
+	}
+	time.Sleep(3 * time.Second)
```

Comment:
why do we need to wait for 3 seconds here? can you comment: make sure operator does not panic after a reasonable amount of time?